webapp/src/Service/DOMJudgeService.php

```diff
@@ -409,11 +410,24 @@ public function getUpdates(): array
             ->setParameter('status', 'open')
             ->getQuery()->getResult();
 
+        $balloons = $this->em->createQueryBuilder()
+            ->select('b.balloonid', 't.name', 't.room', 'p.name', 'cp.color')
+            ->from(Balloon::class, 'b')
+            ->leftJoin('b.submission', 's')
+            ->leftJoin('s.problem', 'p')
+            ->leftJoin('s.contest', 'co')
+            ->leftJoin('p.contest_problems', 'cp', Join::WITH, 'co.cid = cp.contest AND p.probid = cp.problem')
+            ->leftJoin('s.team', 't')
+            ->andWhere('co.cid = :cid')
+            ->setParameter(':cid', $contest->getCid())
```

Comment:
Just a remark: this is now limited to the current contest which seems fine because it mimicks what Clarifications menu and the balloon page itself do.